test/integration/acme/orders_controller_test.go

```diff
@@ -258,11 +259,15 @@ func TestAcmeOrdersController(t *testing.T) {
 			return true, nil
 		}
 		return false, nil
-	})
+	}, timeoutCtx.Done())
 	switch {
 	case err == nil:
 		t.Fatalf("Expected Order to have pending status instead got: %v", pendingOrder.Status.State)
 	case err == wait.ErrWaitTimeout:
+		if ctx.Err() != nil {
+			t.Error(ctx.Err())
+		}
```

Comment:
In this test we want the `timeoutCtx` to timeout. To make sure the `timeoutCtx` was not canceled due to `ctx` timing-out, this check is added.

Reply:
Ideally, this logic is replaced with a more elegant solution, there is already a TODO comment above this section.

Reply:
I see. So we expect `timeoutCtx` to timeout, and exit the PollImmediate function, but it might also have exited because the parent "test-wide" context has timed out, which would have Cancelled the `timeoutCtx`, so we'd get a test error here saying "context cancelled: timeout" or something like that.